### DISCUSSION TITLE
cannot run fooocus

### DISCUSSION THREAD

**Discussion by michaelyeung445** [lllyasviel/Fooocus#3408] [Q&A] [answered]
I started to run the 2.5.2version for the first time, and appear this message:
 import supervision as sv
ModuleNotFoundError: No module named 'supervision'

How to fix this

**Comment by Rogala** [comment]
[[Fixed] ModuleNotFoundError: No module named ‘supervision’](https://blog.finxter.com/fixed-modulenotfounderror-no-module-named-supervision/)

**Comment by mashb1t** [comment]
Please follow the "how to update" section in the [2.5.0 release notes](https://github.com/lllyasviel/Fooocus/releases/tag/v2.5.0).

**Reply by michaelyeung445** [reply]
still not working

**Reply by mashb1t** [reply]
What's not working, do you have another error or can you provide any terminal logs/screenshots?

**Comment by michaelyeung445** [comment]
sorry, I mean the same message after follow the instruction:
import supervision as sv
ModuleNotFoundError: No module named 'supervision'

**Reply by mashb1t** [reply]
please provie the full terminal logs, from first line to last line. Thanks!

**Reply by michaelyeung445** [reply]
C:\Users\Micha\OneDrive\Desktop\Fooocus_win64_2-1-831>.\python_embeded\python.exe -s Fooocus\entry_with_update.py --preset realistic
Already up-to-date
Update succeeded.
[System ARGV] ['Fooocus\\entry_with_update.py', '--preset', 'realistic']
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Fooocus version: 2.5.2
Version mismatch for torchsde: Installed version 0.2.5 does not meet requirement torchsde==0.2.6
Installing requirements
Couldn't install requirements.
Command: "C:\Users\Micha\OneDrive\Desktop\Fooocus_win64_2-1-831\python_embeded\python.exe" -m pip install -r "requirements_versions.txt" --prefer-binary
Error code: 1
stdout: Collecting torchsde==0.2.6 (from -r requirements_versions.txt (line 1))
  Obtaining dependency information for torchsde==0.2.6 from https://files.pythonhosted.org/packages/dd/1f/b67ebd7e19ffe259f05d3cf4547326725c3113d640c277030be3e9998d6f/torchsde-0.2.6-py3-none-any.whl.metadata
  Using cached torchsde-0.2.6-py3-none-any.whl.metadata (5.3 kB)
Collecting einops==0.8.0 (from -r requirements_versions.txt (line 2))
  Obtaining dependency information for einops==0.8.0 from https://files.pythonhosted.org/packages/44/5a/f0b9ad6c0a9017e62d4735daaeb11ba3b6c009d69a26141b258cd37b5588/einops-0.8.0-py3-none-any.whl.metadata
  Using cached einops-0.8.0-py3-none-any.whl.metadata (12 kB)
Collecting transformers==4.42.4 (from -r requirements_versions.txt (line 3))
  Obtaining dependency information for transformers==4.42.4 from https://files.pythonhosted.org/packages/6a/dc/23c26b7b0bce5aaccf2b767db3e9c4f5ae4331bd47688c1f2ef091b23696/transformers-4.42.4-py3-none-any.whl.metadata
  Using cached transformers-4.42.4-py3-none-any.whl.metadata (43 kB)
Collecting safetensors==0.4.3 (from -r requirements_versions.txt (line 4))
  Obtaining dependency information for safetensors==0.4.3 from https://files.pythonhosted.org/packages/28/78/82c03572b085b658d24a13ae0ae4779c0712a167af6c313d39ee5b5e7f73/safetensors-0.4.3-cp310-none-win_amd64.whl.metadata
  Using cached safetensors-0.4.3-cp310-none-win_amd64.whl.metadata (3.9 kB)
Collecting accelerate==0.32.1 (from -r requirements_versions.txt (line 5))
  Obtaining dependency information for accelerate==0.32.1 from https://files.pythonhosted.org/packages/e4/74/564f621699b049b0358f7ad83d7437f8219a5d6efb69bbfcca328b60152f/accelerate-0.32.1-py3-none-any.whl.metadata
  Using cached accelerate-0.32.1-py3-none-any.whl.metadata (18 kB)
Collecting pyyaml==6.0.1 (from -r requirements_versions.txt (line 6))
  Obtaining dependency information for pyyaml==6.0.1 from https://files.pythonhosted.org/packages/24/97/9b59b43431f98d01806b288532da38099cc6f2fea0f3d712e21e269c0279/PyYAML-6.0.1-cp310-cp310-win_amd64.whl.metadata
  Using cached PyYAML-6.0.1-cp310-cp310-win_amd64.whl.metadata (2.1 kB)
Collecting pillow==10.4.0 (from -r requirements_versions.txt (line 7))
  Obtaining dependency information for pillow==10.4.0 from https://files.pythonhosted.org/packages/f4/72/0203e94a91ddb4a9d5238434ae6c1ca10e610e8487036132ea9bf806ca2a/pillow-10.4.0-cp310-cp310-win_amd64.whl.metadata
  Using cached pillow-10.4.0-cp310-cp310-win_amd64.whl.metadata (9.3 kB)
Collecting scipy==1.14.0 (from -r requirements_versions.txt (line 8))
  Obtaining dependency information for scipy==1.14.0 from https://files.pythonhosted.org/packages/a3/0c/82c1330c08f31d61142d38cb9a185e01c2403c990d10dab208032e62d0fa/scipy-1.14.0-cp310-cp310-win_amd64.whl.metadata
  Using cached scipy-1.14.0-cp310-cp310-win_amd64.whl.metadata (60 kB)
Collecting tqdm==4.66.4 (from -r requirements_versions.txt (line 9))
  Obtaining dependency information for tqdm==4.66.4 from https://files.pythonhosted.org/packages/18/eb/fdb7eb9e48b7b02554e1664afd3bd3f117f6b6d6c5881438a0b055554f9b/tqdm-4.66.4-py3-none-any.whl.metadata
  Using cached tqdm-4.66.4-py3-none-any.whl.metadata (57 kB)
Collecting psutil==6.0.0 (from -r requirements_versions.txt (line 10))
  Obtaining dependency information for psutil==6.0.0 from https://files.pythonhosted.org/packages/73/44/561092313ae925f3acfaace6f9ddc4f6a9c748704317bad9c8c8f8a36a79/psutil-6.0.0-cp37-abi3-win_amd64.whl.metadata
  Using cached psutil-6.0.0-cp37-abi3-win_amd64.whl.metadata (22 kB)
Collecting pytorch_lightning==2.3.3 (from -r requirements_versions.txt (line 11))
  Obtaining dependency information for pytorch_lightning==2.3.3 from https://files.pythonhosted.org/packages/cf/1c/6122684e93864c09cf8010e27ad82b3b2f4864bd6864a40cb9e040752277/pytorch_lightning-2.3.3-py3-none-any.whl.metadata
  Using cached pytorch_lightning-2.3.3-py3-none-any.whl.metadata (21 kB)
Collecting omegaconf==2.3.0 (from -r requirements_versions.txt (line 12))
  Obtaining dependency information for omegaconf==2.3.0 from https://files.pythonhosted.org/packages/e3/94/1843518e420fa3ed6919835845df698c7e27e183cb997394e4a670973a65/omegaconf-2.3.0-py3-none-any.whl.metadata
  Using cached omegaconf-2.3.0-py3-none-any.whl.metadata (3.9 kB)
Requirement already satisfied: gradio==3.41.2 in c:\users\micha\onedrive\desktop\fooocus_win64_2-1-831\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 13)) (3.41.2)
Collecting pygit2==1.15.1 (from -r requirements_versions.txt (line 14))
  Obtaining dependency information for pygit2==1.15.1 from https://files.pythonhosted.org/packages/16/27/0310ab41371bb64e149e50d954aa089b140f269159d4d6ed34c83e4be7f2/pygit2-1.15.1-cp310-cp310-win_amd64.whl.metadata
  Using cached pygit2-1.15.1-cp310-cp310-win_amd64.whl.metadata (3.4 kB)
Collecting opencv-contrib-python-headless==4.10.0.84 (from -r requirements_versions.txt (line 15))
  Obtaining dependency information for opencv-contrib-python-headless==4.10.0.84 from https://files.pythonhosted.org/packages/01/d4/dacf890940cb22279e1513b7ea41d97825a723153a5efce68bc52e7b3b6e/opencv_contrib_python_headless-4.10.0.84-cp37-abi3-win_amd64.whl.metadata
  Using cached opencv_contrib_python_headless-4.10.0.84-cp37-abi3-win_amd64.whl.metadata (20 kB)
Collecting httpx==0.27.0 (from -r requirements_versions.txt (line 16))
  Obtaining dependency information for httpx==0.27.0 from https://files.pythonhosted.org/packages/41/7b/ddacf6dcebb42466abd03f368782142baa82e08fc0c1f8eaa05b4bae87d5/httpx-0.27.0-py3-none-any.whl.metadata
  Using cached httpx-0.27.0-py3-none-any.whl.metadata (7.2 kB)
Collecting onnxruntime==1.18.1 (from -r requirements_versions.txt (line 17))
  Obtaining dependency information for onnxruntime==1.18.1 from https://files.pythonhosted.org/packages/74/8e/0db236760a60ba59fac48e253c2c645bfd7d6ade58d4ce796b645d73e2de/onnxruntime-1.18.1-cp310-cp310-win_amd64.whl.metadata
  Using cached onnxruntime-1.18.1-cp310-cp310-win_amd64.whl.metadata (4.5 kB)
Collecting timm==1.0.7 (from -r requirements_versions.txt (line 18))
  Obtaining dependency information for timm==1.0.7 from https://files.pythonhosted.org/packages/fe/85/834c70052b518bb8fca457d8ea4e60a65d9cc41f77fd409eeff3b3041638/timm-1.0.7-py3-none-any.whl.metadata
  Using cached timm-1.0.7-py3-none-any.whl.metadata (47 kB)
Collecting numpy==1.26.4 (from -r requirements_versions.txt (line 19))
  Obtaining dependency information for numpy==1.26.4 from https://files.pythonhosted.org/packages/19/77/538f202862b9183f54108557bfda67e17603fc560c384559e769321c9d92/numpy-1.26.4-cp310-cp310-win_amd64.whl.metadata
  Using cached numpy-1.26.4-cp310-cp310-win_amd64.whl.metadata (61 kB)
Collecting tokenizers==0.19.1 (from -r requirements_versions.txt (line 20))
  Obtaining dependency information for tokenizers==0.19.1 from https://files.pythonhosted.org/packages/f4/85/d999b9a05fd101d48f1a365d68be0b109277bb25c89fb37a389d669f9185/tokenizers-0.19.1-cp310-none-win_amd64.whl.metadata
  Using cached tokenizers-0.19.1-cp310-none-win_amd64.whl.metadata (6.9 kB)
Collecting packaging==24.1 (from -r requirements_versions.txt (line 21))
  Obtaining dependency information for packaging==24.1 from https://files.pythonhosted.org/packages/08/aa/cc0199a5f0ad350994d660967a8efb233fe0416e4639146c089643407ce6/packaging-24.1-py3-none-any.whl.metadata
  Using cached packaging-24.1-py3-none-any.whl.metadata (3.2 kB)
Collecting rembg==2.0.57 (from -r requirements_versions.txt (line 22))
  Obtaining dependency information for rembg==2.0.57 from https://files.pythonhosted.org/packages/0c/29/74b542899b19d33761b5fbb6a625b3dcdfd946966f7615c8d80567d29d81/rembg-2.0.57-py3-none-any.whl.metadata
  Using cached rembg-2.0.57-py3-none-any.whl.metadata (16 kB)
Collecting groundingdino-py==0.4.0 (from -r requirements_versions.txt (line 23))
  Using cached groundingdino-py-0.4.0.tar.gz (82 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'

stderr:   error: subprocess-exited-with-error

  python setup.py egg_info did not run successfully.
  exit code: 1

  [6 lines of output]
  Traceback (most recent call last):
    File "<string>", line 2, in <module>
    File "<pip-setuptools-caller>", line 34, in <module>
    File "C:\Users\Micha\AppData\Local\Temp\pip-install-hn_e704g\groundingdino-py_fad692344826449faecc3f4b6c93fe5c\setup.py", line 41, in <module>
      readme = readme_file.read()
  UnicodeDecodeError: 'cp950' codec can't decode byte 0xf0 in position 2876: illegal multibyte sequence
  [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

Encountered error while generating package metadata.

See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

[notice] A new release of pip is available: 23.2.1 -> 24.2
[notice] To update, run: C:\Users\Micha\OneDrive\Desktop\Fooocus_win64_2-1-831\python_embeded\python.exe -m pip install --upgrade pip

CMD Failed requirements: install -r "requirements_versions.txt"
Loaded preset: C:\Users\Micha\OneDrive\Desktop\Fooocus_win64_2-1-831\Fooocus\presets\realistic.json
Failed to load config key: {"path_checkpoints": ["C:\\Users\\Micha\\Downloads\\Fooocus_win64_2-1-831\\Fooocus\\models\\checkpoints"]} is invalid or does not exist; will use {"path_checkpoints": ["../models/checkpoints/"]} instead.
Failed to load config key: {"path_loras": ["C:\\Users\\Micha\\Downloads\\Fooocus_win64_2-1-831\\Fooocus\\models\\loras"]} is invalid or does not exist; will use {"path_loras": ["../models/loras/"]} instead.
Failed to load config key: {"path_embeddings": "C:\\Users\\Micha\\Downloads\\Fooocus_win64_2-1-831\\Fooocus\\models\\embeddings"} is invalid or does not exist; will use {"path_embeddings": "../models/embeddings/"} instead.
Failed to load config key: {"path_vae_approx": "C:\\Users\\Micha\\Downloads\\Fooocus_win64_2-1-831\\Fooocus\\models\\vae_approx"} is invalid or does not exist; will use {"path_vae_approx": "../models/vae_approx/"} instead.
Failed to load config key: {"path_vae": "C:\\Users\\Micha\\Downloads\\Fooocus_win64_2-1-831\\Fooocus\\models\\vae"} is invalid or does not exist; will use {"path_vae": "../models/vae/"} instead.
Failed to load config key: {"path_upscale_models": "C:\\Users\\Micha\\Downloads\\Fooocus_win64_2-1-831\\Fooocus\\models\\upscale_models"} is invalid or does not exist; will use {"path_upscale_models": "../models/upscale_models/"} instead.
Failed to load config key: {"path_inpaint": "C:\\Users\\Micha\\Downloads\\Fooocus_win64_2-1-831\\Fooocus\\models\\inpaint"} is invalid or does not exist; will use {"path_inpaint": "../models/inpaint/"} instead.
Failed to load config key: {"path_controlnet": "C:\\Users\\Micha\\Downloads\\Fooocus_win64_2-1-831\\Fooocus\\models\\controlnet"} is invalid or does not exist; will use {"path_controlnet": "../models/controlnet/"} instead.
Failed to load config key: {"path_clip_vision": "C:\\Users\\Micha\\Downloads\\Fooocus_win64_2-1-831\\Fooocus\\models\\clip_vision"} is invalid or does not exist; will use {"path_clip_vision": "../models/clip_vision/"} instead.
Failed to load config key: {"path_fooocus_expansion": "C:\\Users\\Micha\\Downloads\\Fooocus_win64_2-1-831\\Fooocus\\models\\prompt_expansion\\fooocus_expansion"} is invalid or does not exist; will use {"path_fooocus_expansion": "../models/prompt_expansion/fooocus_expansion"} instead.
Failed to load config key: {"path_wildcards": "C:\\Users\\Micha\\Downloads\\Fooocus_win64_2-1-831\\Fooocus\\wildcards"} is invalid or does not exist; will use {"path_wildcards": "../wildcards/"} instead.
Failed to load config key: {"path_safety_checker": "C:\\Users\\Micha\\Downloads\\Fooocus_win64_2-1-831\\Fooocus\\models\\safety_checker"} is invalid or does not exist; will use {"path_safety_checker": "../models/safety_checker/"} instead.
[Cleanup] Attempting to delete content of temp dir C:\Users\Micha\AppData\Local\Temp\fooocus
[Cleanup] Cleanup successful
Total VRAM 8192 MB, total RAM 32534 MB
Set vram state to: NORMAL_VRAM
Always offload VRAM
Device: cuda:0 NVIDIA GeForce RTX 3060 Ti : native
VAE dtype: torch.bfloat16
Traceback (most recent call last):
  File "C:\Users\Micha\OneDrive\Desktop\Fooocus_win64_2-1-831\Fooocus\entry_with_update.py", line 46, in <module>
    from launch import *
  File "C:\Users\Micha\OneDrive\Desktop\Fooocus_win64_2-1-831\Fooocus\launch.py", line 147, in <module>
    from webui import *
  File "C:\Users\Micha\OneDrive\Desktop\Fooocus_win64_2-1-831\Fooocus\webui.py", line 10, in <module>
    import modules.async_worker as worker
  File "C:\Users\Micha\OneDrive\Desktop\Fooocus_win64_2-1-831\Fooocus\modules\async_worker.py", line 3, in <module>
    from extras.inpaint_mask import generate_mask_from_image, SAMOptions
  File "C:\Users\Micha\OneDrive\Desktop\Fooocus_win64_2-1-831\Fooocus\extras\inpaint_mask.py", line 6, in <module>
    from extras.GroundingDINO.util.inference import default_groundingdino
  File "C:\Users\Micha\OneDrive\Desktop\Fooocus_win64_2-1-831\Fooocus\extras\GroundingDINO\util\inference.py", line 9, in <module>
    import supervision as sv
ModuleNotFoundError: No module named 'supervision'

C:\Users\Micha\OneDrive\Desktop\Fooocus_win64_2-1-831>pause
請按任意鍵繼續 . . .

**Reply by mashb1t** [reply, marked as answer]
Please run this command: `C:\Users\Micha\OneDrive\Desktop\Fooocus_win64_2-1-831\python_embeded\python.exe -m pip install -r C:\Users\Micha\OneDrive\Desktop\Fooocus_win64_2-1-831\Fooocus\requirements_versions.txt`

Alternatively you can redownload Fooocus using the windows zip file installation method in the readme.

**Reply by michaelyeung445** [reply]
run on cmd?

**Reply by michaelyeung445** [reply]
I tried your command but still appear the import supervision as sv
ModuleNotFoundError: No module named 'supervision'
